### ISSUE TITLE
comment on Scope class is not accurate regarding span ended

### ISSUE THREAD

**Issue by brucedlg** [open-telemetry/opentelemetry-cpp#3109]
`api/include/opentelemetry/trace/scope.h` has following comment on the Scope class:

```
/**
 * Controls how long a span is active.
 *
 * On creation of the Scope object, the given span is set to the currently
 * active span. On destruction, the given span is ended and the previously
 * active span will be the currently active span again.
 */
class Scope final
```

It specifies that "the given span is ended". This is not accurate. The span is not ended simply because scope object is destroyed.

Users may not explicitly call span.End() based on the comment. This could delay the end of span.

**Comment by lalitb** [comment]
This is correct. On the destruction of `scope` object, the currently active span is removed from the active context, so it is no longer active.  The span only ends when either:
 - Span:End() is explicitly called
 - all shared references to the span are released (triggering the span’s destructor).
